### PR TITLE
improvement: add a flag to allow python to generate discriminated unions as undiscriminated unions

### PR DESCRIPTION
**Pull request by armandobelardo**
This PR adds a config for API settings (`use-undiscriminated-unions-with-literals`) that creates discriminated unions as undiscriminated unions, forcing the discriminant as a literal property on the members

This PR also allows the OpenAPI parsing/conversion step to now take into account the generation language

### CODE REVIEW

File: packages/cli/workspace-loader/src/types/Workspace.ts [hunk 1]
```diff
@@ -36,7 +36,9 @@ export interface Spec {
 
 export interface SpecImportSettings {
     audiences: string[];
+    sdkLanguage: generatorsYml.GenerationLanguage | undefined;
```
Comment (dsinghvi) [inline packages/cli/workspace-loader/src/types/Workspace.ts:39]: why do we add the sdk language here ? we should just pass it from the generator invocation

File: packages/cli/workspace-loader/src/types/Workspace.ts [hunk 2]
```diff
@@ -36,7 +36,9 @@ export interface Spec {
...
     shouldUseTitleAsName: boolean;
+    shouldUseUndiscriminatedUnionsForDiscriminated: boolean;
```
Comment (dsinghvi) [inline packages/cli/workspace-loader/src/types/Workspace.ts:41]: thoughts on a better name here? just brutal for a user to add and we actually need to roll this out broadly at vocode, vellum, all new demos etc

File: packages/cli/configuration/src/generators-yml/schemas/APIConfigurationSchema.ts [hunk 3]
```diff
@@ -14,6 +14,11 @@ export const APIDefinitionSettingsSchema = z.object({
         .optional(z.boolean())
         .describe(
             "Whether to use the titles of the schemas within an OpenAPI definition as the names of the types within Fern. Defaults to true."
+        ),
+    "use-undiscriminated-unions-with-literals": z
```
Comment (dsinghvi) [inline packages/cli/configuration/src/generators-yml/schemas/APIConfigurationSchema.ts:18]: nevermind this looks like a reasonable name

Comment (dsinghvi) [reply packages/cli/configuration/src/generators-yml/schemas/APIConfigurationSchema.ts:18]: might be good to just use this across the whole codebase

File: packages/cli/openapi-parser/src/schema/SchemaParserContext.ts [hunk 4]
```diff
@@ -6,11 +7,20 @@ export interface SchemaParserContext {
...
     shouldUseTitleAsName: boolean;
 
     referenceExists(ref: string): boolean;
     resolveSchemaReference(schema: OpenAPIV3.ReferenceObject): OpenAPIV3.SchemaObject;
     markSchemaAsReferencedByNonRequest(schemaId: SchemaId): void;
     markSchemaAsReferencedByRequest(schemaId: SchemaId): void;
     markReferencedByDiscriminatedUnion(schema: OpenAPIV3.ReferenceObject, discrminant: string, times: number): void;
+
+    shouldUseUndiscriminatedUnionsForDiscriminated: boolean;
+    getShouldUseUndiscriminatedUnionsForDiscriminated(): boolean;
```
Comment (dsinghvi) [inline packages/cli/openapi-parser/src/schema/SchemaParserContext.ts:20]: nit: why do we need both of these functions and not just one?

File: packages/cli/openapi-parser/src/schema/SchemaParserContext.ts [hunk 5]
```diff
@@ -6,11 +7,20 @@ export interface SchemaParserContext {
...
+    storeDiscriminatedUnionMetadata(
+        schema: OpenAPIV3.ReferenceObject,
+        discrminant: string,
+        discriminantValue: string
+    ): void;
```
Comment (dsinghvi) [inline packages/cli/openapi-parser/src/schema/SchemaParserContext.ts:25]: thoughts on making this very specific? 
`markSchemaWithDiscriminant()`  and add docs that we will append literal values here

File: packages/cli/openapi-parser/src/schema/SchemaParserContext.ts (hunk 4 above)
Comment (armandobelardo) [reply packages/cli/openapi-parser/src/schema/SchemaParserContext.ts:20]: ones the flag and the other is a function which evals if it should beused (really just checks the flag as well as the language)

File: packages/cli/cli/src/cli.ts [hunk 6]
```diff
@@ -872,20 +878,27 @@ function addWriteDefinitionCommand(cli: Argv<GlobalCliOptions>, cliContext: CliC
...
         async (argv) => {
             cliContext.instrumentPostHogEvent({
                 command: "fern write-definition"
             });
             await writeDefinitionForWorkspaces({
                 project: await loadProjectAndRegisterWorkspacesWithContext(cliContext, {
                     commandLineApiWorkspace: argv.api,
-                    defaultToAllApiWorkspaces: true
+                    defaultToAllApiWorkspaces: true,
+                    sdkLanguage: argv.language
```
Comment (dsinghvi) [inline packages/cli/cli/src/cli.ts:898]: should we delete `sdkLanguage` from here now that its no longer reequired

**Comment by dsinghvi** [comment]
@armandobelardo cloning this now to play around, will merge after

**Pull request merged**